iothub_client/src/iothub_client_core.c

```diff
@@ -514,69 +526,132 @@ static void iothub_ll_device_twin_callback(DEVICE_TWIN_UPDATE_STATE update_state
+    {
+        LogError("queue_context NULL");
+    }
+    else
     {
         USER_CALLBACK_INFO queue_cb_info;
         queue_cb_info.type = CALLBACK_TYPE_DEVICE_TWIN;
         queue_cb_info.userContextCallback = queue_context->userContext;
-        queue_cb_info.iothub_callback.dev_twin_cb_info.update_state = update_state;
-        queue_cb_info.iothub_callback.dev_twin_cb_info.userCallback = queue_context->userCallback.getTwin;
+        queue_cb_info.iothub_callback.dev_twin_cb_info.updateState = updateState;
+        queue_cb_info.iothub_callback.dev_twin_cb_info.userCallback.getTwin = queue_context->userCallback.getTwin;
         queue_cb_info.iothub_callback.dev_twin_cb_info.userContext = queue_context->userContext;
+        queue_cb_info.iothub_callback.dev_twin_cb_info.twinResponse = NULL;
 
-        if (payLoad == NULL)
+        if (payload == NULL)
         {
-            queue_cb_info.iothub_callback.dev_twin_cb_info.payLoad = NULL;
+            queue_cb_info.iothub_callback.dev_twin_cb_info.payload = NULL;
             queue_cb_info.iothub_callback.dev_twin_cb_info.size = 0;
         }
         else
         {
-            queue_cb_info.iothub_callback.dev_twin_cb_info.payLoad = (unsigned char*)malloc(size);
+            queue_cb_info.iothub_callback.dev_twin_cb_info.payload = (unsigned char*)malloc(size);
 
-            if (queue_cb_info.iothub_callback.dev_twin_cb_info.payLoad == NULL)
+            if (queue_cb_info.iothub_callback.dev_twin_cb_info.payload == NULL)
             {
                 LogError("Failure allocating payload in get device twin callback.");
                 queue_cb_info.iothub_callback.dev_twin_cb_info.size = 0;
             }
             else
             {
-                (void)memcpy(queue_cb_info.iothub_callback.dev_twin_cb_info.payLoad, payLoad, size);
+                (void)memcpy(queue_cb_info.iothub_callback.dev_twin_cb_info.payload, payload, size);
                 queue_cb_info.iothub_callback.dev_twin_cb_info.size = size;
             }
         }
 
+        // Will copy queue_cb_info into vector.  (Not a vector of pointers, but a vector of USER_CALLBACK_INFOs)
         if (VECTOR_push_back(queue_context->iotHubClientHandle->saved_user_callback_list, &queue_cb_info, 1) != 0)
         {
             LogError("device twin callback userContextCallback vector push failed.");
 
-            if (queue_cb_info.iothub_callback.dev_twin_cb_info.payLoad != NULL)
+            if (queue_cb_info.iothub_callback.dev_twin_cb_info.payload != NULL)
             {
-                free(queue_cb_info.iothub_callback.dev_twin_cb_info.payLoad);
+                free(queue_cb_info.iothub_callback.dev_twin_cb_info.payload);
             }
         }
 
         free(queue_context);
     }
+}
+
+static void iothub_ll_get_device_twin_section_async_callback(DEVICE_TWIN_UPDATE_STATE updateState, IOTHUB_TWIN_RESPONSE_HANDLE twinResponse, const unsigned char* payload, size_t size, void* userContextCallback)
```

Comment:
userContextCallback -> userContext ?

Reply:
I'm changing them all to userContext for consistency